File: web/app/Flight.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Flight extends Model
{
  protected $fillable=['aircraft_id', 'flight_status_id'];

  public function flight_status(){
    return $this->hasOne('App\FlightStatus', 'id', 'flight_status_id');
  }

  public function aircraft(){
    return $this->hasOne('App\Aircraft', 'id', 'aircraft_id');
  }

  public function schedule(){
    return $this->hasOne('App\Schedule');
  }

  public function flight_seat(){
    return $this->hasMany('App\FlightSeat');
  }

  public function flight_seat_prices(){
    return $this->hasMany('App\FlightSeatPrice');
  }

}

File: android/app/src/main/java/tut/flightbookingsystem/model/Flight.java
package tut.flightbookingsystem.model;

import java.io.Serializable;
import java.util.List;

public class Flight implements Serializable {
    public long id;
    public long aircraft_id;
    public long flight_status_id;
    public Aircraft aircraft;
    public Schedule schedule;
    public List<FlightSeatPrice> flight_seat_prices;
    public String created_at;
    public String updated_at;


    public double getStartPrice() {
        for (final FlightSeatPrice sP : flight_seat_prices) {
            if (sP.flight_seat.travel_class.name.equals("First")) {
                return sP.price;
            }
        }

        return 0.0;
    }
}
